include/lsst/afw/math/BoundedField.h

```diff
@@ -175,8 +175,16 @@ class BoundedField : public table::io::PersistableFacade<BoundedField>, public t
         return (*this)*(1.0/scale);
     }
 
+    /// BoundedFields (of the same sublcass) are equal if their bounding boxes and parameters are equal.
+    virtual bool operator==(BoundedField const& rhs) const = 0;
+    /// @copydoc operator==
+    bool operator!=(BoundedField const& rhs) const { return !(*this == rhs); };
+
     virtual ~BoundedField() {}
 
+    friend std::ostream & operator<<(std::ostream & os, BoundedField const & bf)
+    { return os << bf.getPersistenceName() << " on " << bf.getBBox(); }
```

Comment:
I think it would make sense to give `BoundedField` an `operator<<` with a polymorphic implementation that this could delegate to.